llvm/lib/Target/SystemZ/SystemZISelLowering.cpp

```diff
@@ -444,6 +446,11 @@ SystemZTargetLowering::SystemZTargetLowering(const TargetMachine &TM,
       setOperationAction(ISD::SRL, VT, Custom);
       setOperationAction(ISD::ROTL, VT, Custom);
 
+      // Add ISD::VECREDUCE_ADD as custom in order to implement
+      // it with VZERO+VSUM
+      if (Subtarget.hasVector()) {
```

Comment:
I don't think we need this check - this whole block is executed only if the vector type is "legal", which pre-supposes "hasVector".

Reply:
ok, i will remove the check.